cdp_scrapers/scraper_utils.py

```diff
@@ -66,6 +90,227 @@ def str_simplified(input_str: str) -> str:
+    ----------
+    name: str
+
+    Returns
+    -------
+    name: Set[str]
+        Full form for name if found, e.g. Thomas for Tom
+
+    References
+    ----------
+    NAME_FULL_FORMS_FILE (btn_givennames_synonyms.txt)
+    is available on https://www.behindthename.com/api/
+    under CC BY-SA 4.0 license (https://creativecommons.org/licenses/by-sa/4.0/).
+    The file is used without modification.
+    """
+    try:
+        with open(NAME_FULL_FORMS_FILE) as name_file:
+            for line in name_file:
+                if re.search(f"^{name}", line, re.IGNORECASE) is not None:
+                    # Tom\tmf\tMaas,Tam,Thom,Thomas,Tomas,Tommie,Tommy
+                    return set(line.split()[-1].split(","))
+
+    except OSError:
+        # file not found
+        pass
+    return set()
+
+
+def name_full_forms(name: str) -> Set[str]:
+    """
+    Return all common expanded variations of name.
+    List is scraped from behindthename.com
+
+    Parameters
+    ----------
+    name: str
+
+    Returns
+    -------
+    names: Set[str]
+        e.g. Thomas, Tomas for Tom
+
+    See Also
+    --------
+    https://www.behindthename.com/name/{name}/related
+    """
+    global known_name_full_forms
+
+    try:
+        # reuse if work was already done for this name
+        return known_name_full_forms[name.lower()]
+    except KeyError:
+        pass
+
+    # prefer names file over API; API is quota limited
+    full_forms = file_search_full_form_names(name)
+    if not full_forms:
+        full_forms = query_full_form_names(name)
+    full_forms |= set([name])
+
+    known_name_full_forms[name.lower()] = full_forms
+    return full_forms
+
+
+def is_same_person(name: str, query_name: str, check_reverse: bool = True) -> bool:
+    """
+    Return True if name and query_name are determined to be same
+
+    Parameters
+    ----------
+    name: str
+        Variations of name are compared against query_name
+    query_name: str
+        This name is fixed during comparisons
+    check_reverse: bool = True
+        Additionally test with name and query_name swapped
+        if initial test fails
+
+    Returns
+    -------
+    bool
+        True if name == query_name or fuzzy matching passes
+
+    Notes
+    -----
+    FuzzyWuzzy's token_sort_ratio() and Metaphone's doublemetaphone
+    are used to compare the names
+    """
+    # for better fuzzy logic, keep just lowercase alphabets and single whitespaces
+    name = to_alphabets_only(name).lower()
+    query_name = to_alphabets_only(query_name).lower()
+
+    # don't waste time if obvious
+    if name == query_name:
+        return True
+    if len(name) == 0 or len(query_name) == 0:
+        return False
+
+    # can't always use the first substring as first name
+    # e.g. first_initial middle_name last_name
+    name_parts = name.split()
+    query_name_parts = query_name.split()
+
+    for i in range(len(name_parts)):
+        if len(name_parts[i]) <= 1 and len(name_parts) <= 2:
```

Comment:
`for i, name_part in enumerate(name_parts):` is more pythonic

Reply:
Ah, thank you!